### PR TITLE
feat: Concurrent emote removals

### PR DESCRIPTION
**Pull request by brian6932**

### CODE REVIEW

File: src/commands/remove/index.ts [hunk 1]
```diff
@@ -25,35 +25,66 @@ registerCommand<PreHandlers>({
...
+			})
+		)).filter(Boolean)
+
+		type out = {
+			Success: boolean
+			Result: string
 		}
 
-		return {
+		return (function (this: out) {
```
Comment (melon095) [inline src/commands/remove/index.ts]: Could you figure out  something that doesn't require this weird function thing or that call method please.

You can remove the out type, and instead use [this](https://github.com/JoachimFlottorp/Melonbot/blob/main/src/Models/Command.ts#L50)

File: src/commands/remove/index.ts [hunk 2]
```diff
@@ -25,35 +25,66 @@ registerCommand<PreHandlers>({
 			this.EarlyEnd.InvalidInput('No emote name provided');
 		}
 
-		const emote = (await gql.CurrentEnabledEmotes(EmoteSet())).find(
-			(emote) => emote.name === ctx.input[0],
-		);
+		ctx.input = ctx.input.reduce((args: string[], arg: string) => {
```
Comment (melon095) [inline src/commands/remove/index.ts]: Unnecessary to add the argument types here.

**Review by melon095 — CHANGES_REQUESTED** [review]

File: src/commands/remove/index.ts (hunk 1 above)
Comment (brian6932) [reply src/commands/remove/index.ts]: I mean ts doesn't complain, it's better than the alternatives, it's more or less how object literals actually work under the hood, while allowing you the freedom of conditions, a `new function()` directly would be better/preferred, but ts was complaining.
I don't believe there is a better option tbh.

File: src/commands/remove/index.ts (hunk 2 above)
Comment (brian6932) [reply src/commands/remove/index.ts]: image_78
Uhh you sure about that?

Comment (melon095) [reply src/commands/remove/index.ts]: Toy language

**Review by melon095 — APPROVED** [review]

**Pull request merged**